src/field_theory/adjoin.lean

```diff
@@ -825,8 +825,8 @@ begin
     { obtain ⟨y, h⟩ := this.mul_inv_cancel hx',
       exact (congr_arg (∈ S1 ⊔ S2) (eq_inv_of_mul_right_eq_one (subtype.ext_iff.mp h))).mp y.2 } },
   refine is_field_of_is_integral_of_is_field' _ (field.to_is_field K),
-  have h1 : algebra.is_algebraic K E1 := algebra.is_algebraic_of_finite,
-  have h2 : algebra.is_algebraic K E2 := algebra.is_algebraic_of_finite,
+  have h1 := @algebra.is_algebraic_of_finite K E1 _ _ _ _,
+  have h2 := @algebra.is_algebraic_of_finite K E2 _ _ _ _,
```

Comment:
What's going on here?

Reply:
This fixes a timeout along the following lines:
```lean
set_option profiler true

lemma slow {K L : Type*} [field K] [field L] [algebra K L]
  (E1 : intermediate_field K L) [h1 : finite_dimensional K E1] :
  algebra.is_algebraic K E1 :=
algebra.is_algebraic_of_finite -- 850ms

lemma fast {K L : Type*} [field K] [field L] [algebra K L]
  (E1 : intermediate_field K L) [h1 : finite_dimensional K E1] :
  algebra.is_algebraic K E1 :=
by apply algebra.is_algebraic_of_finite -- 150ms
```

I'll see if there's a neater way to fix it.

Reply:
I got so frustrated at minor issues with `is_algebraic_iff_integral` that I decided to make a whole PR addressing them (incidentally fixing the timeout): #11773